### PR TITLE
dns-google: fix condition to don't use private dns zones

### PR DESCRIPTION
**Pull request by paulojmdias**
## Pull Request Checklist

- [X] The Certbot team has recently expressed interest in reviewing a PR for this. If not, this PR may be closed due our limited resources and need to prioritize how we spend them.
- [X] If the change being made is to a [distributed component](https://certbot.eff.org/docs/contributing.html#code-components-and-layout), edit the `master` section of `certbot/CHANGELOG.md` to include a description of the change being made.
- [X] Add or update any documentation as needed to support the changes in this PR.
- [X] Include your name in `AUTHORS.md` if you like.

## Description

Validate only `privateVisibilityConfig` is not enough since any user can have a private DNS zone created with the same TLD as a public one and not (yet) attached to any network for some reason.

Below I share a Google API response to improve the validation changes.

```json
❯ curl -H "$(oauth2l header --scope userinfo.email,cloud-dns)" https://dns.googleapis.com/dns/v1/projects/redacted-project/managedZones\?dnsName\=redacted.domain.com.\&alt\=json
{
  "managedZones": [
    {
      "name": "redacted-name-1",
      "dnsName": "redacted.domain.com.",
      "description": "redacted.domain.com-private",
      "id": "REDACTED",
      "nameServers": [
        "ns-gcp-private.googledomains.com."
      ],
      "creationTime": "2023-08-07T14:32:06.894Z",
      "visibility": "private",
      "labels": {},
      "cloudLoggingConfig": {
        "kind": "dns#managedZoneCloudLoggingConfig"
      },
      "kind": "dns#managedZone"
    },
    {
      "name": "redacted-name-2",
      "dnsName": "redacted.domain.com.",
      "description": "redacted.domain.com-public",
      "id": "REDACTED",
      "nameServers": [
        "ns-cloud-e1.googledomains.com.",
        "ns-cloud-e2.googledomains.com.",
        "ns-cloud-e3.googledomains.com.",
        "ns-cloud-e4.googledomains.com."
      ],
      "creationTime": "2023-08-04T12:47:18.680Z",
      "visibility": "public",
      "labels": {},
      "cloudLoggingConfig": {
        "kind": "dns#managedZoneCloudLoggingConfig"
      },
      "kind": "dns#managedZone"
    }
  ],
  "kind": "dns#managedZonesListResponse"
}
```

### CODE REVIEW

**Comment by paulojmdias** [comment]
@alexzorin @ohemorange could someone take a look ? 🙏

**Comment by paulojmdias** [comment]
@bmw could you please take a look? I don't know who can take a look on this PR.

File: certbot-dns-google/certbot_dns_google/_internal/dns_google.py [hunk 1]
```diff
@@ -306,7 +306,7 @@ def _find_managed_zone_id(self, domain: str) -> str:
 
             for zone in zones:
                 zone_id = zone['id']
-                if 'privateVisibilityConfig' not in zone:
+                if zone.get('privateVisibilityConfig') is not None or zone['visibility'] != "private": # pylint: disable=line-too-long
```
Comment (adferrand) [inline certbot-dns-google/certbot_dns_google/_internal/dns_google.py:309]: Why not just forget about the check on `privateVisibilityConfig` ? This property only makes sense for a zone with its `visibility` set to `private`, and this is what we have to test here anyway.

**Review by adferrand — CHANGES_REQUESTED** [review]

File: certbot-dns-google/certbot_dns_google/_internal/dns_google.py (hunk 1 above)
Comment (paulojmdias) [reply certbot-dns-google/certbot_dns_google/_internal/dns_google.py]: Make total sense @adferrand. Updated accordingly

File: certbot-dns-google/certbot_dns_google/_internal/tests/dns_google_test.py [hunk 2]
```diff
@@ -88,6 +88,7 @@ class GoogleClientTest(unittest.TestCase):
     record_ttl = 42
     zone = "ZONE_ID"
     change = "an-id"
+    visibility = "public"
```
Comment (adferrand) [inline certbot-dns-google/certbot_dns_google/_internal/tests/dns_google_test.py:91]: One last thing and I think we are good to go. Could you craft a test (or edit one) where for given two zones for the same tld, one private and one public, the assertions validate that the plugin will take the public one ? You should put the private first in the list to ensure that the behavior is not driven by list ordering.

**Review by adferrand — CHANGES_REQUESTED** [review]

File: certbot-dns-google/certbot_dns_google/_internal/tests/dns_google_test.py (hunk 2 above)
Comment (paulojmdias) [reply certbot-dns-google/certbot_dns_google/_internal/tests/dns_google_test.py:91]: Make sense @adferrand.

I have changed the evaluation behavior to get explicitly only zones with  `public` value on `visibility`. We don't expect changes on `forwarding` and `peering` DNS zones on `visibility` but I think it's better to get only the `public` ones.

I have also added the following [commit](https://github.com/certbot/certbot/pull/9744#:~:text=add%20test%20for%20split%20horizon%20dns%20google) tests for the Split Horizon DNS zones. I haven't found any test using `dnsName` on zone definition, so I have done the same using only the `id` and made the zone with `private` `visibility` appear first on the array.

**Review by adferrand — APPROVED** [review]
This LGTM ! Thanks a lot !

**Comment by bmw** [comment]
Thanks for grabbing the review here Adrien.

**Pull request merged**